src/rgw/rgw_gc.cc

```diff
@@ -128,11 +128,146 @@ int RGWGC::list(int *index, string& marker, uint32_t max, bool expired_only, std
   return 0;
 }
 
-int RGWGC::process(int index, int max_secs)
+class RGWGCIOManager {
+  CephContext *cct;
+  RGWGC *gc;
+
+  struct IO {
+    enum Type {
+      UnknownIO = 0,
+      TailIO = 1,
+      IndexIO = 2,
+    } type{UnknownIO};
+    librados::AioCompletion *c{nullptr};
+    string oid;
+    int index{-1};
+    string tag;
+
+    public:
+    IO() = default;
+
+    IO(IO::Type const type, librados::AioCompletion* const c, 
+       std::string const& oid, int const index, std::string const& tag)
+     : type(type), c(c), oid(oid), index(index), tag(tag)
```

Comment:
style nits: use of const with pass-by-value is overly verbose, and putting the const after the type name is not consistent with the rest of the codebase. the [style guide](https://google.github.io/styleguide/cppguide.html#Use_of_const) discusses this issue, and concludes with:

> That said, while we encourage putting const first, we do not require it. But be consistent with the code around you!

Reply:
I promised someone that I'd give "East const" a go, but I'm ok with "West const", which is indeed the most frequent guest. I agree that it's a bit verbose-- I personally like the explicit communication of intent, but here it accomplishes little other than satisfying my pedanticism. Will update.